units/src/amount/signed.rs

```diff
@@ -63,7 +63,34 @@ impl SignedAmount {
     pub const MAX: Self = SignedAmount::MAX_MONEY;
 
     /// Constructs a new [`SignedAmount`] with satoshi precision and the given number of satoshis.
-    pub const fn from_sat(satoshi: i64) -> SignedAmount { SignedAmount(satoshi) }
+    ///
+    /// # Errors
+    ///
+    /// On values exceeding [`SignedAmount::MAX`] or less than [`SignedAmount::MIN`]
+    pub const fn from_sat(satoshi: i64) -> Result<SignedAmount, OutOfRangeError> {
+        if satoshi < Self::MIN.0 {
+            Err(OutOfRangeError { is_signed: true, is_greater_than_max: false })
+        } else if satoshi > Self::MAX.0 {
+            Err(OutOfRangeError { is_signed: true, is_greater_than_max: true })
+        } else {
+            Ok(SignedAmount(satoshi))
```

Comment:
This `OutOfRangeError` is a bit hard to work with but I believe we can use
```rust
        if satoshi < Self::MIN.0 {
            Err(OutOfRangeError::too_small())
        } else if satoshi > Self::MAX.0 {
            Err(OutOfRangeError::too_big(true))
        } else {
            Ok(SignedAmount(satoshi))
        }
```

Reply:
Oh boy, I see we don't use the helpers in other places. Feel free to ignore my suggestion.